### PR TITLE
add: HelmRepository auth with referenced secret

### PR DESCRIPTION
**Pull request by LittleFox94**
If a HelmRepository references a secret in the database, it will now be used to authenticate at that repository.

The code is a bit clunky, maybe we should add the database to the context instead of passing it around as yet another argument? Built this way to match the already existing code.

### CODE REVIEW

**Comment by raffis** [comment]
upport for it was explicitly removed. Same reason as values with secrets are not supported https://github.com/DoodleScheduling/flux-build/blob/master/internal/build/helm.go#L350-L351.

The reason is usually Secrets are not kept in the gitops repositories as it's insecure and bad practice. Usually Secrets are represented for example as bitnamis SealedSecrets which can only be decrypted on the cluster.

That said I do see the need for this but I wonder about the use case here, do you have your repository secrets in the gitops repos?

**Comment by LittleFox94** [comment]
> Support for it was explicitly removed. Same reason as values with secrets are not supported https://github.com/DoodleScheduling/flux-build/blob/master/internal/build/helm.go#L350-L351.

Ah, I thought it was a "not yet implemented" - a comment would have been useful^^'

We do in fact have secrets in our GitOps repositories, but using sops instead of SealedSecrets. All our secrets are encrypted for a given cluster and for a set of people which again is determined by cluster.

I'm also working on some code to try decrypting sops'ed secrets, replacing each data key with a dummy value if that fails - this way you will at least see a specific secret being used as a specific value and how that propagates into the built HelmRelease

Anyway, having a way to authenticate at the HelmRepository would be really useful - right now flux-build just cannot build HelmReleases with a repository requiring authentication

File: internal/build/helm.go [hunk 1]
```diff
@@ -386,20 +386,35 @@ func (h *Helm) composeValues(ctx context.Context, db map[ref]*resource.Resource,
...
+		obj, _, err := h.opts.Decoder.Decode(raw, nil, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		return obj.(*corev1.Secret), nil
 	}
-	return &secret, nil*/
+
+	return nil, fmt.Errorf("Secret %v/%v not in our db", repository.ObjectMeta.Namespace, repository.Spec.SecretRef.Name)
```
Comment (raffis) [inline internal/build/helm.go:417]:
```suggestion
	return nil, fmt.Errorf("no repository secret `%v` found for helmrepository `%s/%s`", lookupRef, repository.Namespace, repository.Name)
```

**Review by raffis — COMMENTED** [review]
It's fine to merge this, just one small change.

It's probably wrong any way not to support plain secrets directly. I think If something relies on a decrypted secret directly like a HelmRepository which is needed in ci the way to do it likely is feed in that secret manifest before the flux-build execution. For example it may be stored in ci secrets.

File: internal/build/helm.go (hunk 1 above)
Comment (LittleFox94) [reply internal/build/helm.go]: manually applied this suggestion, thanks

**Review by raffis — APPROVED** [review]
Thanks for the contribution 🥇

**Pull request merged**